tests/ui/features/lib/TextEditorPage.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+* ownCloud
+*
+* @author Phillip Davis
+* @copyright 2017 Phillip Davis info@individual-it.net
+*
+* This library is free software; you can redistribute it and/or
+* modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or any later version.
+*
+* This library is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+*
+* You should have received a copy of the GNU Affero General Public
+* License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+*
+*/
+
+namespace Page;
+
+use SensioLabs\Behat\PageObjectExtension\PageObject\Exception\ElementNotFoundException;
+use WebDriver\Key;
+
+class TextEditorPage extends FilesPage
+{
+	protected $newTextFileButtonXpath = './/div[contains(@class, "newFileMenu")]//a[@data-templatename="New text file.txt"]';
+	protected $newTextFileNameInputLabel = 'New text file.txt';
+	protected $newTextFileNameInputXpath = ".//div[contains(@class, \"newFileMenu\")]//a[@data-templatename=\"New text file.txt\"]//input";
+	protected $textFileEditXpath = "//textarea[contains(@class,'ace_text-input')]";
+	protected $textEditorCloseButtonId = "editor_close";
+
+	/**
+	 * type in the field that matches the given xpath and press enter.
+	 * Note: this depends on methods that might only be in the Selenium implementation
```

Comment:
I tried to use keyPress, keyUp, keyDown with no joy.
Using postValue works fine, but it is part of the underlying webDriver that does not seem to get exposed by Mink.
Is there some other way to "just type stuff" where the cursor is, without clearing the current contents of the field?

Reply:
yes the keypressing functions are somehow rubbish, had a lot of trouble with them.
you could try to implement raw typing with JS, or somehow manipulating that field with JS.

Reply:
or I could leave this as it is. and we admit that we depend on Selenium?

Reply:
👍 yes, if you are happy with that solution, just pass in the Session variable from higher up and all is fine. We anyway depend on Selenium